### DISCUSSION TITLE
no matches found: httpx[http2]

### DISCUSSION THREAD

**Discussion by teathinker** [encode/httpx#1656] [Potential Issue] [answered]
I am not sure why pip cannot find http2, the default httpx only works fine. Thanks. I am using python3.6 and/or python3.9

pip install httpx[http2]
zsh: no matches found: httpx[http2]

**Comment by teathinker** [comment, marked as answer]
just need to quote the package name... never mind...

**Reply by lovelydinosaur** [reply]
Ah, that's interesting tho. Is that specifically a `zsh` thing, that square brackets need to be quoted?
We might (or might not) want to think about our docs in that area, if there's a gotcha there for some specific types of shell usage.

**Reply by Kludex** [reply]
There's no loss on quoting in the documentation. I've seen many projects doing it, so it's not a bad idea for us to do the same.

References:
- https://stackoverflow.com/a/30539963 
- https://zsh.sourceforge.io/Guide/zshguide05.html#l137

**Reply by lovelydinosaur** [reply]
Okay.
Would accept a pull request for this.
Would need to change the `README`, and `docs/index.md`.

**Comment by sha2fiddy** [comment]
fyi I struggled for some time trying to solve this issue when running:
`pip install httpx[http2]`
and:
`noglob pip install httpx[http2]`

I was unaware that stringifying packages was allowed, but ultimately this worked for me:
`pip install 'httpx[http2]'`

It might be worth adding a note about this in the http2/optional package section of the docs
